### DISCUSSION TITLE
Using `--filter` in CI

### DISCUSSION THREAD

**Discussion by thezimmee** [vercel/turborepo#1582] [Help] [answered]
Hello! First off, just getting started with it but love the tool, think it'll be heads above lerna. I am, however, having issues with the `--filter` option in my Drone CI. My goal is to "run {x} task for all changes from the `main` branch". Which works as expected locally, but not in my CI with the same build/same commands/same environment. To illustrate...

_Here is output on my local machine of `git diff` and the `turbo --filter` commands. **This is working as expected**:_

![CleanShot 2022-07-26 at 14 36 49@2x](https://user-images.githubusercontent.com/976368/181107668-ff8ddb1d-9aeb-40cc-b9a2-7cb043a48f9f.png)

You can see that the `turbo --filter` command lines up well with the `git diff` output. And indeed, in this case turbo is filtering as I would have expected.

_Now here is output from my Drone CI pipeline. **This is NOT working as expected.** This is the same build as my local build, but it's not detecting the same diff/changes._ You'll also notice in my Drone job I ran the `turbo --filter` command so many different ways, but none of them detected any changes, even though the same commands work locally.

![CleanShot 2022-07-26 at 14 26 28@2x](https://user-images.githubusercontent.com/976368/181107811-a6b7a55d-81d9-4583-901a-79ab3413c721.png)

So I only see the issue in my Drone CI pipeline. Didn't want to file an issue yet, because thinking I might be missing some configuration or something, but no idea where to start and couldn't find answers searching the repo (best I found is https://github.com/vercel/turborepo/issues/1032 and https://github.com/vercel/turborepo/issues/1079, but didn't seem quite the same).

**Any ideas where I might start to figure this out?**

**Comment by tomdavidson** [comment]
Same here with  turbo v1.4.7 and pnpm v7.12.2

**Reply by tomdavidson** [reply]
@thezimmee this seems like a bug or did you get it to work somehow?

**Comment by mehulkar** [comment]
This looks like it could be a bug, can you open an issue instead of a discussion?

**Comment by nathanhammond** [comment, marked as answer]
The issue is that you are passing `--filter` to `npm` on CI, and to Turborepo locally. You need an argument separator in there:

`npm run test -- --filter=[origin/main]`

**Reply by tomdavidson** [reply]
Not the case for me. I was running turbo directly and not as an npm run script.

**Reply by nathanhammond** [reply]
Can you get me a reproduction with more details filed as an issue? The information you have provided so far is not enough for me to do anything with.

**Comment by yevhen-logosha** [comment]
Even better here.. Left(locally) is correctly identify 1 package changed since master.. right(ci) has same output all of the time

<img width="1452" alt="Screenshot 2023-03-20 at 12 59 19" src="https://user-images.githubusercontent.com/25623699/226745026-7e4a0df0-5b30-4158-8c35-de2221a79c73.png">
